In [1]:
import pandas as pd
import numpy as np

In [14]:
dataset = pd.read_csv("Titanic-Dataset.csv")
dataset = dataset.drop(["PassengerId", "Name", "Ticket", "Cabin"], axis=1)
dataset.head(10)

Survived  Pclass     Sex   Age  SibSp  Parch     Fare Embarked
0         0       3    male  22.0      1      0   7.2500        S
1         1       1  female  38.0      1      0  71.2833        C
2         1       3  female  26.0      0      0   7.9250        S
3         1       1  female  35.0      1      0  53.1000        S
4         0       3    male  35.0      0      0   8.0500        S
5         0       3    male   NaN      0      0   8.4583        Q
6         0       1    male  54.0      0      0  51.8625        S
7         0       3    male   2.0      3      1  21.0750        S
8         1       3  female  27.0      0      2  11.1333        S
9         1       2  female  14.0      1      0  30.0708        C

In [15]:
X = dataset.iloc[:,1:].values
y = dataset.iloc[:,0:1].values

In [16]:
print(X)
print("\n")
print(y)

[[3 'male' 22.0 ... 0 7.25 'S']
 [1 'female' 38.0 ... 0 71.2833 'C']
 [3 'female' 26.0 ... 0 7.925 'S']
 ...
 [3 'female' nan ... 2 23.45 'S']
 [1 'male' 26.0 ... 0 30.0 'C']
 [3 'male' 32.0 ... 0 7.75 'Q']]


[[0]
 [1]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [1]
 [1]
 [0]
 [0]
 [0]
 [1]
 [0]
 [1]
 [0]
 [1]
 [0]
 [1]
 [1]
 [1]
 [0]
 [1]
 [0]
 [0]
 [1]
 [0]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [1]
 [1]
 [0]
 [1]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [1]
 [1]
 [0]
 [1]
 [1]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [1]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]


In [17]:
from sklearn.impute import SimpleImputer
imputer = SimpleImputer(missing_values=np.nan, strategy='mean')
imputer.fit(X[:,2:3])
X[:,2:3] = imputer.transform(X[:,2:3])

In [18]:
print(X)

[[3 'male' 22.0 ... 0 7.25 'S']
 [1 'female' 38.0 ... 0 71.2833 'C']
 [3 'female' 26.0 ... 0 7.925 'S']
 ...
 [3 'female' 29.69911764705882 ... 2 23.45 'S']
 [1 'male' 26.0 ... 0 30.0 'C']
 [3 'male' 32.0 ... 0 7.75 'Q']]


In [19]:
X.shape

(891, 7)

In [20]:
print(X[:,6:])

[['S']
 ['C']
 ['S']
 ['S']
 ['S']
 ['Q']
 ['S']
 ['S']
 ['S']
 ['C']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['Q']
 ['S']
 ['S']
 ['C']
 ['S']
 ['S']
 ['Q']
 ['S']
 ['S']
 ['S']
 ['C']
 ['S']
 ['Q']
 ['S']
 ['C']
 ['C']
 ['Q']
 ['S']
 ['C']
 ['S']
 ['C']
 ['S']
 ['S']
 ['C']
 ['S']
 ['S']
 ['C']
 ['C']
 ['Q']
 ['S']
 ['Q']
 ['Q']
 ['C']
 ['S']
 ['S']
 ['S']
 ['C']
 ['S']
 ['C']
 ['S']
 ['S']
 ['C']
 ['S']
 ['S']
 ['C']
 [nan]
 ['S']
 ['S']
 ['C']
 ['C']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['C']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['Q']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['C']
 ['C']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['S']
 ['Q']
 ['S']
 ['C']
 ['S']
 ['S']
 ['C']
 ['S']
 ['Q']
 ['S']
 ['C']
 ['S']
 ['S']
 ['S']
 ['C']
 ['S']
 ['S']
 ['C']
 ['Q']
 ['S']
 ['C']
 ['S']
 ['C']
 ['S']
 ['S']
 ['S']
 ['S']
 ['C']
 ['S']
 ['S']
 ['S']
 ['C']
 ['C']
 ['S']
 ['S']

In [21]:
from sklearn.compose import ColumnTransformer
from sklearn.preprocessing import OneHotEncoder
ct = ColumnTransformer(transformers=[('encoder',OneHotEncoder(),[6])],remainder='passthrough')
X = ct.fit_transform(X)

In [22]:
X

array([[0.0, 0.0, 1.0, ..., 1, 0, 7.25],
       [1.0, 0.0, 0.0, ..., 1, 0, 71.2833],
       [0.0, 0.0, 1.0, ..., 0, 0, 7.925],
       ...,
       [0.0, 0.0, 1.0, ..., 1, 2, 23.45],
       [1.0, 0.0, 0.0, ..., 0, 0, 30.0],
       [0.0, 1.0, 0.0, ..., 0, 0, 7.75]], dtype=object)

In [23]:
X.shape

(891, 10)

In [26]:
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()
X[:,5:6] = le.fit_transform(X[:,5:6]).reshape(-1, 1)

a:\Tensorflow\lib\site-packages\sklearn\preprocessing\_label.py:110: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


	Survived	Pclass	Sex	Age	SibSp	Parch	Fare	Embarked
0	0	3	male	22.0	1	0	7.2500	S
1	1	1	female	38.0	1	0	71.2833	C
2	1	3	female	26.0	0	0	7.9250	S
3	1	1	female	35.0	1	0	53.1000	S
4	0	3	male	35.0	0	0	8.0500	S
5	0	3	male	NaN	0	0	8.4583	Q
6	0	1	male	54.0	0	0	51.8625	S
7	0	3	male	2.0	3	1	21.0750	S
8	1	3	female	27.0	0	2	11.1333	S
9	1	2	female	14.0	1	0	30.0708	C


In [32]:

for x in X[:,0:]:
    print(x)

[0.0 0.0 1.0 0.0 3 1 22.0 1 0 7.25]
[1.0 0.0 0.0 0.0 1 0 38.0 1 0 71.2833]
[0.0 0.0 1.0 0.0 3 0 26.0 0 0 7.925]
[0.0 0.0 1.0 0.0 1 0 35.0 1 0 53.1]
[0.0 0.0 1.0 0.0 3 1 35.0 0 0 8.05]
[0.0 1.0 0.0 0.0 3 1 29.69911764705882 0 0 8.4583]
[0.0 0.0 1.0 0.0 1 1 54.0 0 0 51.8625]
[0.0 0.0 1.0 0.0 3 1 2.0 3 1 21.075]
[0.0 0.0 1.0 0.0 3 0 27.0 0 2 11.1333]
[1.0 0.0 0.0 0.0 2 0 14.0 1 0 30.0708]
[0.0 0.0 1.0 0.0 3 0 4.0 1 1 16.7]
[0.0 0.0 1.0 0.0 1 0 58.0 0 0 26.55]
[0.0 0.0 1.0 0.0 3 1 20.0 0 0 8.05]
[0.0 0.0 1.0 0.0 3 1 39.0 1 5 31.275]
[0.0 0.0 1.0 0.0 3 0 14.0 0 0 7.8542]
[0.0 0.0 1.0 0.0 2 0 55.0 0 0 16.0]
[0.0 1.0 0.0 0.0 3 1 2.0 4 1 29.125]
[0.0 0.0 1.0 0.0 2 1 29.69911764705882 0 0 13.0]
[0.0 0.0 1.0 0.0 3 0 31.0 1 0 18.0]
[1.0 0.0 0.0 0.0 3 0 29.69911764705882 0 0 7.225]
[0.0 0.0 1.0 0.0 2 1 35.0 0 0 26.0]
[0.0 0.0 1.0 0.0 2 1 34.0 0 0 13.0]
[0.0 1.0 0.0 0.0 3 0 15.0 0 0 8.0292]
[0.0 0.0 1.0 0.0 1 1 28.0 0 0 35.5]
[0.0 0.0 1.0 0.0 3 0 8.0 3 1 21.075]
[0.0 0.0 1.0 0.0 3 0 38.0 1 5 31.387

In [33]:
y

array([[0],
       [1],
       [1],
       [1],
       [0],
       [0],
       [0],
       [0],
       [1],
       [1],
       [1],
       [1],
       [0],
       [0],
       [0],
       [1],
       [0],
       [1],
       [0],
       [1],
       [0],
       [1],
       [1],
       [1],
       [0],
       [1],
       [0],
       [0],
       [1],
       [0],
       [0],
       [1],
       [1],
       [0],
       [0],
       [0],
       [1],
       [0],
       [0],
       [1],
       [0],
       [0],
       [0],
       [1],
       [1],
       [0],
       [0],
       [1],
       [0],
       [0],
       [0],
       [0],
       [1],
       [1],
       [0],
       [1],
       [1],
       [0],
       [1],
       [0],
       [0],
       [1],
       [0],
       [0],
       [0],
       [1],
       [1],
       [0],
       [1],
       [0],
       [0],
       [0],
       [0],
       [0],
       [1],
       [0],
       [0],
       [0],
       [1],
       [1],
       [0],
       [1],
       [1],
    

In [36]:
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.2)

In [38]:
print(X_train)
print("\n")
print(y_train)
print("\n")
print(X_test)
print("\n")
print(y_test)
print("\n")

[[0.0 0.0 1.0 ... 1 1 14.4]
 [0.0 0.0 1.0 ... 0 0 8.05]
 [1.0 0.0 0.0 ... 0 0 512.3292]
 ...
 [0.0 0.0 1.0 ... 0 1 55.0]
 [0.0 0.0 1.0 ... 4 2 31.275]
 [0.0 0.0 1.0 ... 0 0 10.5]]


[[0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [1]
 [1]
 [0]
 [0]
 [1]
 [0]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [1]
 [0]
 [1]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [0]
 [1]
 [1]
 [0]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [1]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [1]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [1]
 [0]
 [1]
 [0]
 [1]
 [1]
 [0]
 [0]
 [1]
 [1]
 [1]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [0]
 [1]
 [1]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [1]
 [0

In [39]:
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train[:,4:] = sc.fit_transform(X_train[:,4:])
X_test[:,4:] = sc.transform(X_test[:,4:])

In [40]:
X_train
print("\n")
X_test

array([[0.0, 1.0, 0.0, ..., 1.3093976776656826, -0.4707631301819402,
        -0.17847143779905647],
       [0.0, 1.0, 0.0, ..., -0.4579789033731155, -0.4707631301819402,
        -0.4777697329524224],
       [0.0, 0.0, 1.0, ..., -0.4579789033731155, -0.4707631301819402,
        -0.4772869937666912],
       ...,
       [1.0, 0.0, 0.0, ..., -0.4579789033731155, 0.757014703845684,
        0.49213439136483234],
       [0.0, 0.0, 1.0, ..., -0.4579789033731155, -0.4707631301819402,
        -0.47495439802123784],
       [1.0, 0.0, 0.0, ..., 1.3093976776656826, 0.757014703845684,
        -0.2555494381063914]], dtype=object)